lua/dap.lua

```diff
@@ -335,6 +328,39 @@ function Session.event_output(_, body)
   repl.append(body.output, '$')
 end
 
+function Session:_request_scopes(current_frame)
+  self:request('scopes', { frameId = current_frame.id }, function(_, scopes_resp)
+    if not scopes_resp or not scopes_resp.scopes then return end
+
+    current_frame.scopes = {}
+    for _, scope in pairs(scopes_resp.scopes) do
+
+      table.insert(current_frame.scopes, scope)
+      if not scope.expensive then
+        self:request('variables', { variablesReference = scope.variablesReference }, function(_, variables_resp)
+          if not variables_resp then return end
+
+          scope.variables = variables_resp.variables
+        end)
+      end
+    end
+  end)
+end
+
+function Session:_frame_delta(delta)
+  if self.stopped_thread_id and self.threads[self.stopped_thread_id].frames then
+    local frames = self.threads[self.stopped_thread_id].frames
+    local current_frame_index = index_of(frames, function(i) return i.id == self.current_frame.id end)
+
+    current_frame_index = current_frame_index + delta
+    if 0 < current_frame_index and current_frame_index <= #frames then
+      self.current_frame = frames[current_frame_index]
+
+      jump_to_frame(self.current_frame, true)
```

Comment:
@theHamsta btw did you set the `preserve_focus_hint` to `true` here deliberately? For me `.up` / `.down` turns out to be way more useful if it jumps to the location

Reply:
No, not deliberately. At that moment, I actually wanted to have `true` for jump to location. But apparently I have done the opposite.